File: node_depths/node_depths.py
def node_depths(root, depth=0):
  if root is None:
    return 0
  return depth + node_depths(root.left, depth + 1) + node_depths(root.right, depth + 1)

class BST:
  def __init__(self, value):
      self.value = value
      self.left = None
      self.right = None


bst = BST(10)
bst.left = BST(8)
bst.right = BST(12)

print(node_depths(bst))

File: node_depths/node_depths.ts
class BST {
    value: number;
    left: BST | null;
    right: BST | null;

    constructor(value: number) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

function nodeDepths(root: BST | null, depth: number = 0): number {
    if (root === null) {
        return 0;
    }
    return depth + nodeDepths(root.left, depth + 1) + nodeDepths(root.right, depth + 1);
}

const bst = new BST(10);
bst.left = new BST(8);
bst.right = new BST(12);

console.log(nodeDepths(bst));  
